python/pyarrow/tests/test_array.py

```diff
@@ -1499,3 +1499,10 @@ def test_concat_array():
 def test_concat_array_different_types():
     with pytest.raises(pa.ArrowInvalid):
         pa.concat_arrays([pa.array([1]), pa.array([2.])])
+
+
+def test_array_from_strided_numpy_array():
```

Comment:
This should go into `pyarrow/tests/test_pandas.py` along with the existing tests for strided inputs.